closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -261,22 +264,47 @@ static void printErrors(void)
     fflush(stderr);
 }
 
+/*
+ * We use a 8 digit map (AABBCCDD) to represent the version of openssl.
+ * AA is the major version,
+ * BB is the minor version,
+ * CC is the revision, and DD for the letter that could be present in any version.
+ * For example, if an openssl version is in this scheme 1.1.1k.
+ * The ossl_ver should be 10101011 ( k is the 11th letter value for DD here... ).
+ */
+static jlong ver_to_l(const char* astring)
+{
+    long major = 0;
+    long minor = 0;
+    long revision = 0;
+    long diff = 0;
+    int num = sscanf(astring, "OpenSSL %ld.%ld.%ld", &major, &minor, &revision);
+    if (num < 0) {
+        return -1;
+    }
+    char last_char = tolower(astring[strlen(astring) - 1]);
+    if (isalpha(last_char)) {
+        diff = abs(last_char - 'a') + 1;
+    }
+    return (jlong)((major * 10000000L) + (minor * 100000L) + (revision * 1000L) + diff);
+}
+
 static void *crypto_library = NULL;
 /*
  * Class:     jdk_crypto_jniprovider_NativeCrypto
  * Method:    loadCrypto
  * Signature: (Z)I
  */
-JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
   (JNIEnv *env, jclass thisObj, jboolean trace)
 {
     char *error;
     typedef const char* OSSL_version_t(int);
+    long ossl_ver;
```

Comment:
Please leave this declaration in its original location, and initialize to zero.

Reply:
Backed and initialized.